packages/app-wallet-interface/source/index.html.md

```diff
@@ -0,0 +1,321 @@
+---
+title: API Reference
+
+language_tabs: # must be one of https://git.io/vQNgJ
+  - json
+
+toc_footers:
+  - <a href='#'>Sign Up for a Developer Key</a>
+  - <a href='https://github.com/lord/slate'>Documentation Powered by Slate</a>
+
+includes:
+
+search: true
+---
+
+# Introduction
+
+API between the state channel wallet and application.
+
+# DataTypes
+
+## Participant
+
+```json
+{
+  "participantId": "abc123", // allocated by app
+  "signingAddress": "0x123456",
+  "destination": "0xa..."
+}
+```
+
+| Parameter      | Type   | Description                                                     |
+| -------------- | ------ | --------------------------------------------------------------- |
+| participantId  | String | App allocated id, used for relaying messages to the participant |
+| signingAddress | String | Address used to sign channel updates                            |
+| destination    | String | Address of EOA to receive channel proceeds                      |
+
+Note: in the future we might replace the `appId` with a `contactAddress`, which would allow
+apps to get the wallet to relay messages itself. An example `contractAddress` would be something
+like `https://myserver.com/state_channel_callback`.
+
+## Allocation
+
+```json
+{
+  "token": "0x...", // 0x0 for ETH
+  "allocationItems": [
+    {"destination": "0xa...", "amount": "12"},
+    {"destination": "0xb...", "amount": "12"}
+  ]
+}
+```
+
+## Message
+
+Format of message sent from the wallet to the app, so that the app can then relay it
+to another participant.
+
+```javascript
+const participantId = 'abc123'; // allocated by app
+const data = '0x456';
+const message = new Message(participantId, data);
+```
+
+```json
+{
+  "participantId": "abc123",
+  "data": "0x456"
+}
+```
+
+| Parameter     | Type   | Description                                                  |
+| ------------- | ------ | ------------------------------------------------------------ |
+| participantId | String | Identifier or user that the message should be relayed to     |
+| data          | String | Message payload. Format defined by wallet and opaque to app. |
+
+# Initial API
+
+Still a WIP! Adding methods as we need them.
+
+## List identities
+
+## Create Channel
+
+```json
+{
+  "jsonrpc": "2.0",
+  "method": "CreateChannel",
+  "id": 1,
+  "params": {
+    "participants": [
+      {"participantId": "user123", "signingAddress": "0x...", "destination": "0xa..."},
```

Comment:
Is this `signingAddress` gotten from the wallet via another API call? If the app/client is providing it it means the `wallet` can't sign things states on their behalf.

Reply:
Yes. We'll need another API call to get the `signingAddress`.